### PR TITLE
Edit functionality closes #9 closes #10

### LINKED ISSUE #9: Editing a note
When viewing an existing reminder. The user can click on an "Edit" button where they can modify the title, notes, or date of the reminder.

When editing a note, there is a "Save" button that commits the changes to the server. When the user hits the save button, they are returned to the original individual note.

### LINKED ISSUE #10: Refactor new-reminder-form
In #8, we introduced a `new-reminder-form` component that has a lot of code hard coded into it. Let's refactor it into a component that we just called `reminder-form` that we could use for both editing reminders in addition to creating new reminder.

Some research points:

- [Ember Best Practice: Stop Bubbling and Use Closure Actions](https://dockyard.com/blog/2015/10/29/ember-best-practice-stop-bubbling-and-use-closure-actions)
- [Closure Actions in Ember](https://emberway.io/route-closure-actions-in-ember-js-d0a7a37a5d1b#.ko5ueszgy)

### PR DESCRIPTION
**Pull request by kerrd89**
## Purpose

Edit functionality added.

## Approach

User is able to edit any reminder by clicking an edit button on a selected reminder.  When a user clicks save, that update is registered and the user is returned to that updated reminder.

### Learning

Brandon helped us make the form for both new and edited reminders a flexible component. 

Ember docs for creating and updating records.

### Test coverage 

Added an acceptance test which selects the first reminder, clicks the edit button, resets values for that reminder, presses save, and checks that values have been updated correctly each step along the way.  Added to prior tests to confirm the button text changes on the form depending on whether it is being used to create a new reminder or update an existing reminder.

### CODE REVIEW

**Comment by brittanystoroz** [comment]
30 commits 😱

File: app/components/reminder-form.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+});
```
Comment (kerrd89) [inline app/components/reminder-form.js]: Brandon said we had to have that in order for templates/components/reminder-form.hbs.  If we remove that it breaks our app.

Comment (brittanystoroz) [inline app/components/reminder-form.js]: frameworks are dumb 😑

Comment (Jeff-Duke) [inline app/components/reminder-form.js]: What's the screaming emoji for?  Too few commits?

Comment (brittanystoroz) [inline app/components/reminder-form.js]: Too many, it's often really nice to be able to click through each commit and see what changed each step of the way before the PR was submitted. 30 is too many to click through and a lot of them are just conflict resolutions. You might be interested in learning about `git pull --rebase` in the future -- it allows you to do all of your work for the PR, then apply it on top of what already exists in the codebase. This lets you avoid creating commits that simply resolve conflicts.

**Comment by brittanystoroz** [comment]
Actually, this PR looks like it's including commits from previous features. I see commits in this PR from the initial features that were implementing.  The only commits in this PR should be new ones that build on top of master as it currently exists.

File: app/components/reminder-form.js (hunk 1 above)
Comment (Jeff-Duke) [inline app/components/reminder-form.js]: ok we'll look at that.

**Pull request merged**